model/modelprocessor/datastream.go

```diff
@@ -84,3 +74,39 @@ func isRUMAgentName(agentName string) bool {
 	}
 	return false
 }
+
+func metricsetDataset(event *model.APMEvent) string {
+	if event.Transaction != nil || event.Span != nil || event.Service.Name == "" {
+		// Metrics that include well-defined transaction/span fields
+		// (i.e. breakdown metrics, transaction and span metrics) will
+		// be stored separately from custom application metrics.
+		return model.InternalMetricsDataset
+	}
+
+	if event.Metricset != nil {
+		// Well-defined system/runtime metrics are stored in the shared
+		// "internal" metrics data stream, as they are not application-specific.
+		//
+		// TODO(axw) agents should indicate that a metricset is internal.
+		// If a metricset is identified as internal, then we'll ignore any
+		// metrics that we don't already know about; otherwise they will end
+		// up creating service-specific data streams.
+		internal := true
```

Comment:
Are there plans to extend the Intake API to accommodate this?

Reply:
Only in my head for the moment. I have a recurring reminder to create an issue - I'll try to do this in the next couple of days.